### PR TITLE
[Comb] Additional mux canonicalization

### LINKED ISSUE #635: [comb] Canonicalize single bit muxes
Seen in an IR dump, mux of a single bit can often be strength reduced, e.g.:

`comb.mux %5, %171, %false : i1`  can turn into: `comb.and %5, %171`

### PR DESCRIPTION
**Pull request by uenoku**
This is the follow-up to #648.
```
mux(a, false, b) => and(not(a), b)
mux(a, b, true) => or(not(a), b)
```
Will close #635.

### CODE REVIEW

File: test/Dialect/RTL/canonicalization.mlir [hunk 1]
```diff
@@ -804,6 +804,52 @@ rtl.module @mux_canonicalize3(%a: i1, %b: i4) -> (i4) {
   rtl.output %0 : i4
 }
 
+// CHECK-LABEL: rtl.module @mux_canonicalize4(%a: i1, %b: i1) -> (i1) {
+// CHECK-NEXT:   %true = comb.constant true
+// CHECK-NEXT:   %0 = comb.xor %a, %true : i1
+// CHECK-NEXT:   %1 = comb.and %0, %b : i1
+// CHECK-NEXT:  rtl.output %1 : i1
+rtl.module @mux_canonicalize4(%a: i1, %b: i1) -> (i1) {
```
Comment (darthscsi) [inline test/Dialect/RTL/canonicalization.mlir:812]: All the mux cases could be checked in one module.

**Review by darthscsi — APPROVED** [review]
I dub this patch "material implication"

**Review by lattner — APPROVED** [review]
nice!

**Comment by lattner** [comment]
nice!

**Pull request merged**